prober/icmp.go

```diff
@@ -164,22 +202,35 @@ func ProbeICMP(ctx context.Context, target string, module config.Module, registr
 		level.Error(logger).Log("msg", "Error marshalling packet", "err", err)
 		return
 	}
+
 	durationGaugeVec.WithLabelValues("setup").Add(time.Since(setupStart).Seconds())
 	level.Info(logger).Log("msg", "Writing out packet")
 	rttStart := time.Now()
-	if _, err = socket.WriteTo(wb, ip); err != nil {
+	if _, err = socket.WriteTo(wb, dst); err != nil {
 		level.Warn(logger).Log("msg", "Error writing to socket", "err", err)
 		return
 	}
 
-	// Reply should be the same except for the message type.
+	// Reply should be the same except for the message type and ID if the kernel
```

Comment:
For all these comments it would be better if the unprivileged was mentioned, as-is it's a bit confusing.

Reply:
Updated the comments.